tools/rvn/LogTrafficWatch.cs

```diff
@@ -0,0 +1,226 @@
+
+                                builder.Reset();
+                                builder.Renew("trafficwatch/receive", BlittableJsonDocumentBuilder.UsageMode.None);
+
+                                if (await UnmanagedJsonParserHelper.ReadAsync(peepingTomStream, parser, state, buffer).ConfigureAwait(false) == false)
+                                    continue;
+
+                                await UnmanagedJsonParserHelper.ReadObjectAsync(builder, peepingTomStream, parser, buffer).ConfigureAwait(false);
+                                using (var json = builder.CreateReader())
+                                {
+                                    if (_changeTypes != null)
+                                    {
+                                        if (json.TryGetMember("Type", out var obj) == false)
+                                            continue;
+                                        if (obj is LazyStringValue lzv && Enum.TryParse(lzv.ToString(), out TrafficWatchChangeType type))
+                                        {
+                                            if (_changeTypes.Contains(type) == false)
+                                                continue;
+                                        }
+                                        else
+                                        {
+                                            continue;
+                                        }
+                                    }
+
+                                    // log to file
+                                    json.WriteJsonTo(fileStream);
+                                    await fileStream.FlushAsync();
+                                }
+
+                                if (new FileInfo(file).Length >= maxFileSize)
+                                {
+                                    using (var oldFileStream = fileStream)
+                                    {
+                                        var oldFileName = file;
+                                        file = GetFileName();
+
+                                        fileStream = SafeFileStream.Create(file, FileMode.Append, FileAccess.Write, FileShare.Read, 32 * 1024, false);
+
+                                        var t = Task.Run(() =>
+                                        {
+                                            using (var logStream = SafeFileStream.Create(oldFileName, FileMode.Open, FileAccess.Read))
+                                            using (var newFileStream = SafeFileStream.Create(oldFileName + ".gz", FileMode.Create, FileAccess.Write))
+                                            using (var compressionStream = new GZipStream(newFileStream, CompressionMode.Compress))
+                                            {
+                                                logStream.CopyTo(compressionStream);
+                                            }
+
+                                            File.Delete(oldFileName);
```

Comment:
There might be a race here with closing of the old file stream, mind.

Reply:
done